### PR TITLE
[smhi] Add aggregated channels for daily forecast.

### PR DESCRIPTION
**Pull request by pacive**
Also updates to use Optionals instead of `@Nullable`s, and add unit tests

Signed-off-by: Anders Alfredsson <andersb86@gmail.com>

### CODE REVIEW

File: bundles/org.openhab.binding.smhi/src/main/java/org/openhab/binding/smhi/internal/SmhiHandler.java [hunk 1]
```diff
@@ -159,74 +156,87 @@ private void updateChannels(TimeSeries timeSeries) {
...
                 case WIND_DIRECTION:
-                    newState = new QuantityType<>(value, Units.DEGREE_ANGLE);
+                    newState = new QuantityType<>(value.get(), Units.DEGREE_ANGLE);
                     break;
                 case WIND_SPEED:
+                case WIND_MAX:
+                case WIND_MIN:
                 case GUST:
-                    newState = new QuantityType<>(value, Units.METRE_PER_SECOND);
+                    newState = new QuantityType<>(value.get(), SIUnits.METRE.divide(Units.SECOND));
```
Comment (fwolter) [inline bundles/org.openhab.binding.smhi/src/main/java/org/openhab/binding/smhi/internal/SmhiHandler.java:209]: Why did you change this to a custom unit calculation? Same for below.

File: bundles/org.openhab.binding.smhi/src/main/resources/OH-INF/thing/channel-types.xml [hunk 2]
```diff
@@ -8,97 +8,127 @@
 		<item-type>Number:Pressure</item-type>
 		<label>Air Pressure</label>
 		<description>Air pressure in hPa</description>
-		<state readOnly="true" pattern="%.1f %unit%"/>
+		<state readOnly="true" pattern="%.1f hPa"/>
```
Comment (fwolter) [inline bundles/org.openhab.binding.smhi/src/main/resources/OH-INF/thing/channel-types.xml:11]: What's the reason for replacing the units by constants?

**Review by fwolter — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.smhi/src/main/java/org/openhab/binding/smhi/internal/SmhiHandler.java (hunk 1 above)
Comment (pacive) [reply bundles/org.openhab.binding.smhi/src/main/java/org/openhab/binding/smhi/internal/SmhiHandler.java:209]: When testing out the binding, I noticed that the values returned from the Item didn't keep the unit that the QuantityType was created with (some with m/s and some with mm/h), but instead used the default, which is km/h for Number:Speed-Items. Since all other types seemed to work correctly (even the Number:Length-Items where some have km an some mm for unit oddly enough) I thought it had something to to with the predefined composite units, so tried to set it explicitly. Forgot to revert afterwards.

File: bundles/org.openhab.binding.smhi/src/main/resources/OH-INF/thing/channel-types.xml (hunk 2 above)
Comment (pacive) [reply bundles/org.openhab.binding.smhi/src/main/resources/OH-INF/thing/channel-types.xml:11]: This however did solve the above problem, so I figured I'd set it explicitly for all channels, so there's a sensible default. I can still be overridden by setting the state description on the Item to a different unit. If I use %unit% for the channels, the state description would have to be set for each Item to make it display the correct unit, which can be a lot of work for the end user. 

Example:
Without explicit unit:
```
openhab> openhab:status SMHIWeatherForecast_MinWindSpeed
5.759990784011058554102274358886417 km/h
```
With explicit unit:
```
openhab> openhab:status SMHIWeatherForecast_MinWindSpeed
1.6 m/s
```
Should this be considered a bug, and an issue raised in openhab-core?

Comment (fwolter) [reply bundles/org.openhab.binding.smhi/src/main/resources/OH-INF/thing/channel-types.xml:11]: The units might be converted by the framework due to locale settings. As it's unlikely that this binding will be used by users with imperial units, I think it's good to go with constant units.

File: bundles/org.openhab.binding.smhi/src/main/java/org/openhab/binding/smhi/internal/SmhiHandler.java (hunk 1 above)
Comment (pacive) [reply bundles/org.openhab.binding.smhi/src/main/java/org/openhab/binding/smhi/internal/SmhiHandler.java]: Reverted now

**Review by fwolter — APPROVED** [review]
LGTM

As OH3 is already feature-freezed, your PR will be merged as soon as OH3 is released. Your changes will go into OH3.1, then.
